src/atmega328p/com/usart_initialize.rs

```diff
@@ -0,0 +1,458 @@
+//     (at your option) any later version.
+//
+//     This program is distributed in the hope that it will be useful,
+//     but WITHOUT ANY WARRANTY; without even the implied warranty of
+//     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//     GNU Affero General Public License for more details.
+//
+//     You should have received a copy of the GNU Affero General Public License
+//     along with this program.  If not, see <https://www.gnu.org/licenses/>
+
+use crate::atmega328p::hal::gating;
+/// Other source code files to be used
+use crate::atmega328p::hal::interrupt;
+use crate::atmega328p::hal::port;
+
+/// Crates which would be used in the implementation.
+/// We will be using standard volatile and bit_field crates now for a better read and write.
+use crate::delay::delay_ms;
+use bit_field::BitField;
+use core::ptr::write_volatile;
+use core::{f64, u32, u8};
+use volatile::Volatile;
+
+/// Some useful constants regarding bit manipulation for USART.
+/// Position of clock mode adjuster (xck) bit.
+const USART0_XCK: u8 = 4;
+/// System Clock Crystal Oscillator Frequency in mHz.
+const F_OSC: f64 = 1.0000;
+const MULTIPLY: f64 = 1000000.00;
+
+/// Selection of which USART is to be used.
+#[derive(Clone, Copy)]
+pub enum UsartNum {
+    Usart0,
+}
+
+/// Selection of synchronous or asynchronous modes for USART.
+#[derive(Clone, Copy)]
+pub enum UsartModes {
+    Normasync,
+    Douasync,
+    Mastersync,
+    Slavesync,
+}
+
+/// Selection of the parity mode for USART.
+#[derive(Clone, Copy)]
+pub enum UsartParity {
+    No,
+    Even,
+    Odd,
+}
+
+/// Selection of the Amount of Data Bits to be transferred or recieved through USART.
+#[derive(Clone, Copy)]
+pub enum UsartDataSize {
+    Five,
+    Six,
+    Seven,
+    Eight,
+    Nine,
+}
+
+/// Selection of number of stop bits for USART data.
+#[derive(Clone, Copy)]
+pub enum UsartStop {
+    One,
+    Two,
+}
+
+/// Selection of the clock parity mode.
+#[derive(Clone, Copy)]
+pub enum UsartPolarity {
+    Outputrise,
+    Inputrise,
+}
+
+/// This structure contains various registers needed to control usart communication
+/// through ATMEGA320P device.
+/// USART0 is controlled by a total of 6 registers stored through this structure.
+#[repr(C, packed)]
+pub struct Usart {
+    pub ucsra: Volatile<u8>,
+    pub ucsrb: Volatile<u8>,
+    pub ucsrc: Volatile<u8>,
+    _pad: Volatile<u8>, // Padding to look for empty memory space.
+    pub ubrrl: Volatile<u8>,
+    pub ubrrh: Volatile<u8>,
+    pub udr: Volatile<u8>,
+}
+
+/// Various implementation functions for the USART protocol.
+impl Usart {
+    /// This creates a new memory mapped structure of the USART0 for it's control.
+    pub unsafe fn new(num: UsartNum) -> &'static mut Usart {
+        match num {
+            UsartNum::Usart0 => &mut *(0xC0 as *mut Usart),
+        }
+    }
+}
+
+impl Usart {
+    /// Function to disable global interrupts for smooth non-interrupted functioning of USART.
```

Comment:
Dont use `function to` , use `disables ...`

Reply:
corrected everywhere